src/cli.rs

```diff
@@ -5,12 +5,13 @@ use std::{
 
 use clap::{Args, Parser};
 
+const DEFAULT_NODE_RPC_ADDR: SocketAddr =
+    SocketAddr::V4(SocketAddrV4::new(Ipv4Addr::LOCALHOST, 18443));
+
 #[derive(Clone, Args)]
 pub struct NodeRpcConfig {
-    #[arg(default_value = "localhost", long = "node-rpc-host")]
-    pub host: String,
-    #[arg(default_value = "18443", long = "node-rpc-port")]
-    pub port: u16,
+    #[arg(default_value_t = DEFAULT_NODE_RPC_ADDR, long = "node-rpc-addr")]
+    pub addr: SocketAddr,
```

Comment:
This makes it impossible to specify `localhost` as the host. I think that's unfortunate. If we're going to consolidate the option (which makes sense!) this should be a string, or something that at least accepts `localhost:PORT`

Reply:
```
error: invalid value 'localhost:38332' for '--node-rpc-addr <ADDR>': invalid socket address syntax
```

This is rather unfortunate... But not a blocker, do as you want